templates/pages/login.jsx

```diff
@@ -46,7 +49,30 @@ var Login = React.createClass({
     form.processFormData(this.handleFormData);
   },
   handleFormData: function(error, data) {
-    console.log("inside App we see:", error, data);
+    var queryObj = Url.parse(window.location.href, true).query;
+    fetch('/login', {
+      method: 'post',
+      headers: {
+        'Accept': 'application/json; charset=utf-8',
+        'Content-Type': 'application/json; charset=utf-8'
+      },
+      body: JSON.stringify({
+        uid: data.username,
+        password: data.password
+      })
+    }).then(function(response) {
+      var redirectObj;
+      if ( response.status === 200 ) {
+        redirectObj = Url.parse('/login/oauth/authorize', true);
+        redirectObj.query = queryObj;
```

Comment:
I think you also need to set search to null right? From the [docs](https://nodejs.org/api/url.html#url_url_format_urlobj):

```
search will be used in place of query.

    It is treated the same with or without the leading ? (question mark)

query (object; see querystring) will only be used if search is absent.
```